common/ledger/blkstorage/blockindex.go

```diff
@@ -267,6 +305,28 @@ func constructTxIDKey(txID string, blkNum, txNum uint64) []byte {
 	return append(k, util.EncodeOrderPreservingVarUint64(txNum)...)
 }
 
+// decodeTxIDKey takes input an encoded txid key of the format `prefix:len(TxID):TxID:BlkNum:TxNum`
+// and returns the bytes segment `len(TxID):TxID` from this
+func decodeTxIDKey(encodedTxIDKey []byte) ([]byte, error) {
+	if len(encodedTxIDKey) == 0 {
+		return nil, errors.New("Invalid txIDKey - zero-length slice")
+	}
+	if encodedTxIDKey[0] != txIDIdxKeyPrefix {
+		return nil, errors.Errorf("Invalid txIDKey {%x} - unexpected prefix", encodedTxIDKey)
+	}
+	remainingBytes := encodedTxIDKey[utf8.RuneLen(txIDIdxKeyPrefix):]
+
+	txIDLen, n, err := util.DecodeOrderPreservingVarUint64(remainingBytes)
+	if err != nil {
+		return nil, errors.WithMessagef(err, "Invalid txIDKey {%x}", encodedTxIDKey)
+	}
+	remainingBytes = remainingBytes[n:]
```

Comment:
```suggestion
	keyWithoutPrefixAndTxID = keyWithoutPrefix[n:]
```
Makes it longer but worth it as it helps readability.

Reply:
Same as before. How far will you go if you were to write the full function to extract other elements?